src/events/message-reaction-add/message-report.ts

```diff
@@ -0,0 +1,65 @@
+import { EmbedBuilder } from "discord.js";
+import type { MessageReactionAddEvent } from "../../handlers/messageReactionAddEvents.handler";
+
+const reportEmojis = process.env.REPORT_EMOJIS.split(",");
+const reportEmojiAmount = Number(process.env.REPORT_EMOJI_AMOUNT);
+const reportChannelId = process.env.REPORT_CHANNEL_ID;
+
+const event: MessageReactionAddEvent = {
+  execute: async (client, reaction) => {
+    if (!reportEmojis.length || !reportEmojiAmount || !reportChannelId) return;
+    if (reaction.partial) {
+      try {
+        await reaction.fetch();
+      } catch (error) {
+        console.error(
+          "Something went wrong when fetching the message: ",
+          error,
+        );
+        return;
+      }
+    }
+
+    if (
+      reaction.message.author?.bot === true ||
+      reportEmojis.every((emoji) => emoji !== reaction.emoji.name) ||
+      reaction.count === null ||
+      reaction.count !== reportEmojiAmount
```

Comment:
Looks danger to me, should we use `<` instead of `!==` here? Or course if then we will need to delete the reported message once `reaction.count >= reportEmojiAmount`.
```suggestion
      reaction.count < reportEmojiAmount
```

Reply:
I was doing `<`, but since we dont have a db, hence, we can't know the exact previous reported message to deleted

Reply:
I don't understand, can't we directly delete the `reaction.message` right after generate this embed message?

Reply:
My plan was to send the report message without deleting the message

Reply:
its all make sense if we delete it after reported